sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -268,18 +268,27 @@ private[parquet] class ParquetRowConverter(
         }
 
       // For INT32 backed decimals
-      case t: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT32 =>
-        new ParquetIntDictionaryAwareDecimalConverter(t.precision, t.scale, updater)
+      case _: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT32 =>
+        val metadata = parquetType.asPrimitiveType().getDecimalMetadata
+        val precision = metadata.getPrecision()
+        val scale = metadata.getScale()
+        new ParquetIntDictionaryAwareDecimalConverter(precision, scale, updater)
 
       // For INT64 backed decimals
-      case t: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT64 =>
-        new ParquetLongDictionaryAwareDecimalConverter(t.precision, t.scale, updater)
+      case _: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT64 =>
+        val metadata = parquetType.asPrimitiveType().getDecimalMetadata
```

Comment:
decimal metadata can be null, when the data in parquet file is non-decimal (e.g. int) and the user specified schema is decimal type. We should handle it to avoid NPE.

Reply:
Thanks. I'll update soon.